src/index.ts

```diff
@@ -31,31 +31,17 @@ export type NotarizeOptions = NotarizeStartOptions;
 
 export async function startNotarize(opts: NotarizeStartOptions): Promise<NotarizeResult> {
   d('starting notarize process for app:', opts.appPath);
+  var appPath = opts.appPath;
   return await withTempDir<NotarizeResult>(async (dir) => {
-    const zipPath = path.resolve(dir, `${path.basename(opts.appPath, '.app')}.zip`);
-    d('zipping application to:', zipPath);
-    const zipResult = await spawn(
-      'zip',
-      [
-        '-r',
-        '-y',
-        zipPath,
-        path.basename(opts.appPath),
-      ],
-      {
-        cwd: path.dirname(opts.appPath),
-      },
-    );
-    if (zipResult.code !== 0) {
-      throw new Error(`Failed to zip application, exited with code: ${zipResult.code}\n\n${zipResult.output}`);
-    }
-    d('zip succeeded, attempting to upload to apple');
-
+  //zip the app unless it is dmg
+  if ( !opts.appPath.endsWith(`.dmg`) ){
+    appPath = await zipApp(dir, opts.appPath);
+  }
```

Comment:
Maybe something along the lines of

```typescript
if ((await fs.stat(opts.appPath)).isDirectory()) {
  uploadPath = await zipApp(dir, opts.appPath);
} else if (!opts.appPath.endsWith('.dmg')) {
  throw new Error('Path does not resolve to an Electron app');
}
```

Reply:
Does `isDirectory` really return true for an app, thanks apple 😆